jda-commands/src/main/java/com/github/kaktushose/jda/commands/extension/Extension.java

```diff
@@ -0,0 +1,57 @@
+package com.github.kaktushose.jda.commands.extension;
+
+import com.github.kaktushose.jda.commands.JDACommands;
+import com.github.kaktushose.jda.commands.JDACommandsBuilder;
+import com.github.kaktushose.jda.commands.definitions.description.Descriptor;
+import com.github.kaktushose.jda.commands.dispatching.adapter.TypeAdapter;
+import com.github.kaktushose.jda.commands.dispatching.instance.InstanceProvider;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.List;
+
+/// Implementations of this interface, that are registered by java's service provider interface, will be called
+/// in [JDACommandsBuilder] to configure the framework.
+///
+/// Extensions can for example set an implementation of [InstanceProvider] or [Descriptor] or
+/// register additional implementations of [TypeAdapter].
+///
+/// If the implementation of this class needs additional configuration, implementations have to provide an
+/// own implementation of [Data] that the user have to register in the builder before the extensions are loaded.
```

Comment:
```suggestion
/// own implementation of [Data] that the user has to register in the builder before the extensions are loaded.
```
also please link the method that has to be called